### ISSUE TITLE
Dealing with a big dataset in LPRNet.

### ISSUE THREAD

**Issue by farshadopencv** [opencv/openvino_training_extensions#114]
Hi.
I would like to train the LPRNet with a dataset of the size **1.3** million images of  (94,24) plates. I have 3 questions:

1. Is it reasonable to use a dataset with this huge number of plates? Is there a maximum number for the size of datasets for LPRNet?

2. The default values of data augmentation in config file are:

`apply_basic_aug = False
` 
`apply_stn_aug = True
`
 

Now, with 1.3 million images, do I need to use  `apply_basic_aug = TRUE`?

3. How should I change parameters `steps = 250000`  and `learning_rate = 0.001` in order to fit with the **1.3** million number of images? Do I need to  increase the train steps to a higher value like `steps = 450000` for example? In the LPRNet paper, the authors mentioned that they drop the learning rate after every 100k iterations by a factor of 10 and train the network for 250k iterations in total. But I cant find the line for setting the decay factor 10 in config file. Could you please guide me  how to set  hyperparameters `steps`  and `learning_rate` for `1.3` million images dataset? The defaults are:

`batch_size = 32
`
`steps = 250000
`
`learning_rate = 0.001
`
`grad_noise_scale = 0.001
`
`opt_type = 'Adam'
`
 Thanks a lot.

**Comment by AlexanderDokuchaev** [comment]
Hi @farshadopencv 
1. Of course! 
`"We don't have better algorithms. We just have more data.” - Peter Norvig, Chief Scientist, Google.` 
2. It's not necessary but you can try to use both options. May be basic augmentation increase metric in your case..
3. I have no idea why, but `lerning_rate` set in [lpr/tools/train.py#L74](https://github.com/opencv/openvino_training_extensions/blob/develop/tensorflow_toolkit/lpr/tools/train.py#L74). 
Yes, for large dataset you will better if you increase intervals for `lerning_rate` and `steps` value.  

Hard to say about hyperparameters without experiments. In your case I would start by increasing `steps` value and `learning rate policy` in 2 (may be 4) times. Additionally you can try increase `batch_size`.

**Comment by farshadopencv** [comment]
Thanks so much @AlexanderDokuchaev.
